### PR TITLE
Implementation of Model table

### PR DESCRIPTION
**Pull request by avishniakov**
## Describe changes
Here I have added the required functionality to operate with the Model entity in ZenStore.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Review by fa9r — COMMENTED** [review]
Structure looks good overall. I assume all the empty models that just have `pass` will actually hold some data eventually? LMK once they are fully implemented with their actual fields then I'll do another more thorough review

File: src/zenml/model/model_config.py
Comment (avishniakov) [reply src/zenml/model/model_config.py:20]: Placeholder for future, not relevant for this PR actually, I'll kick it out for now

**Comment by avishniakov** [comment]
This PR turns out to be somewhat blocking other work, @fa9r is it good to merge, if tests pass? I will then base my new branches on this one

**Review by fa9r — APPROVED** [review]
@avishniakov Sure, changes so far LGTM.

However, if you still experience the broken env issue then I'd suggest you to investigate first whether this comes from the changes in this PR. I'd also suggest to run a pipeline (ideally quickstart) once to ensure that still works, and to fix potential CI issues before the merge.

Background: We've had quite a few experiences with mega-branches developed in parallel to develop (e.g. see #879) and my personal main takeaway is that you should treat these branches like develop and not like feature branches, i.e., by only pushing / merging non-broken states with passing tests / linting / ... Otherwise you will end up with a feature-complete but totally broken mega-branch in the end that will be hell to fix (talking from experience here 😅)

**Comment by avishniakov** [comment]
> @avishniakov Sure, changes so far LGTM.
> 
> However, if you still experience the broken env issue then I'd suggest you to investigate first whether this comes from the changes in this PR. I'd also suggest to run a pipeline (ideally quickstart) once to ensure that still works, and to fix potential CI issues before the merge.
> 
> Background: We've had quite a few experiences with mega-branches developed in parallel to develop (e.g. see #879) and my personal main takeaway is that you should treat these branches like develop and not like feature branches, i.e., by only pushing / merging non-broken states with passing tests / linting / ... Otherwise you will end up with a feature-complete but totally broken mega-branch in the end that will be hell to fix (talking from experience here 😅)

Thanks @fa9r , yeah I fixed my local - it was changes driven. Absolutely agree, that this FB needs to be treated with respect - will not merge broken there, thus want tests to pass. On top will try with quickstart, as you advised!

**Pull request merged**
